nexus/db-model/src/schema_versions.rs

```diff
@@ -0,0 +1,798 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Database schema versions and upgrades
+//!
+//! For details, see schema/crdb/README.adoc in the root of this repository.
+
+use anyhow::{bail, ensure, Context};
+use camino::Utf8Path;
+use omicron_common::api::external::SemverVersion;
+use once_cell::sync::Lazy;
+use std::collections::BTreeMap;
+
+/// The version of the database schema this particular version of Nexus was
+/// built against
+///
+/// This must be updated when you change the database schema.  Refer to
+/// schema/crdb/README.adoc in the root of this repository for details.
+pub const SCHEMA_VERSION: SemverVersion = SemverVersion::new(45, 0, 0);
+
+/// List of all past database schema versions, in *reverse* order
+///
+/// If you want to change the Omicron database schema, you must update this.
+static KNOWN_VERSIONS: Lazy<Vec<KnownVersion>> = Lazy::new(|| {
+    vec![
+        // +- The next version goes here!  Duplicate this line, uncomment
+        // |  the *second* copy, then update that copy for your version,
+        // |  leaving the first copy as an example for the next person.
+        // v
+        // KnownVersion::new(next_int, "unique-dirname-with-the-sql-files"),
+        KnownVersion::new(45, "migration-demo"),
```

Comment:
Thanks for doing this in a way that encourages "only bumping major version" for now!

As a *super* nitpicky nitpick, can we change the name from `migration-demo` to like `first-named-migration`? At first glance, I thought it was test data, but I realize it actually is a durable long-lived migration.

Reply:
I'll second the super nitpicky nitpick; the name threw me off also. (I was wondering if it was a demo that shoudl be removed before this lands.)